### ISSUE TITLE
optionalJoin+inIfValue behaves unexpectedly in presence of unrelated leftJoin

### ISSUE THREAD

**Issue by lorefnon** [juanluispaz/ts-sql-query#112]
I think there is some inconsistency around how `inIfValue` affects an optional join in presence of other joins. 

Let's say  I have following query: 

```ts
    const conn = getConnection();
    const rows = await conn.selectFrom(tPost)
        .optionalJoin(tPostLocation)
        .on(tPostLocation.postId.equals(tPost.id))
        .where(tPost.slug.equals("smth"))
        .and(tPostLocation.locationId.inIfValue(null))
        .select(tPostCols)
        .executeSelectMany();
    console.log({ rows });                                                                                                                                                         
```

This generates following SQL query: 

```sql
select id as id, slug as slug from post where slug = ? 
```

Notable is that `post_location` table is not joined because `inIfValue` is passed null. 

However, this behavior changes if an unrelated left join is added to the query: 

```ts
    const tPostContentLJ = tPostContent.forUseInLeftJoin();
    const rows = await conn.selectFrom(tPost)
        .leftJoin(tPostContentLJ)  // <------
        .on(tPostContentLJ.postId.equals(tPost.id).and(tPostContentLJ.localeCode.equals("hi-IN")))
        .optionalJoin(tPostLocation)
        .on(tPostLocation.postId.equals(tPost.id))
        .where(tPost.slug.equals("smth"))
        .and(tPostLocation.locationId.inIfValue(null))
        .select(tPostCols)
        .executeSelectMany();
    console.log({ rows });
```

Now the SQL query generated is as follows: 

```sql
select post.id as id, post.slug as slug 
from post 
left join post_content on post_content.post_id = post.id and post_content.locale_code = ? 
join post_location on post_location.post_id = post.id 
where post.slug = ?
```

This is strange because `post_location` is still not queried, however it is included in join. This causes the number of rows selected to change.

This behavior differs from `equalsIfValue`, so if I update the above usage to: `tPostLocation.locationId.equalsIfValue(null)` the `post_location` table is no longer included in join.

**Comment by juanluispaz** [comment]
Fixed in ts-sql-query 1.56.0 already released

**Comment by lorefnon** [comment]
This is great. Thank you 👍🏾
